build.go setup problems with gomodules

This is not very important, more of a reminder:

gometalinter doesn't support gomodules (https://github.com/alecthomas/gometalinter/issues/569):

```
[...]
github.com/alecthomas/gometalinter
go: finding github.com/alecthomas/gometalinter v2.0.12+incompatible
go: downloading github.com/alecthomas/gometalinter v2.0.12+incompatible
go: extracting github.com/alecthomas/gometalinter v2.0.12+incompatible
go: finding github.com/google/shlex latest
go: finding gopkg.in/alecthomas/kingpin.v3-unstable latest
go: downloading github.com/google/shlex v0.0.0-20181106134648-c34317bd91bf
go: extracting github.com/google/shlex v0.0.0-20181106134648-c34317bd91bf
go: downloading gopkg.in/alecthomas/kingpin.v3-unstable v3.0.0-20180810215634-df19058c872c
go: extracting gopkg.in/alecthomas/kingpin.v3-unstable v3.0.0-20180810215634-df19058c872c
go: finding github.com/alecthomas/units latest
go: finding github.com/nicksnyder/go-i18n/i18n latest
go: downloading github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf
go: extracting github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf
go: finding github.com/nicksnyder/go-i18n v1.10.0
go: downloading github.com/nicksnyder/go-i18n v1.10.0
go: extracting github.com/nicksnyder/go-i18n v1.10.0
go: downloading gopkg.in/yaml.v2 v2.0.0-20171116090243-287cf08546ab
go: extracting gopkg.in/yaml.v2 v2.0.0-20171116090243-287cf08546ab
go: finding github.com/pelletier/go-toml v1.2.0
go: downloading github.com/pelletier/go-toml v1.2.0
go: extracting github.com/pelletier/go-toml v1.2.0
# github.com/alecthomas/gometalinter
../../../../pkg/mod/github.com/alecthomas/gometalinter@v2.0.12+incompatible/main.go:36:94: cannot use loadConfig (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("config", "Load JSON configuration from file.").Envar("GOMETALINTER_CONFIG").Action
../../../../pkg/mod/github.com/alecthomas/gometalinter@v2.0.12+incompatible/main.go:38:100: cannot use disableAction (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("disable", "Disable previously enabled linters.").PlaceHolder("LINTER").Short('D').Action
../../../../pkg/mod/github.com/alecthomas/gometalinter@v2.0.12+incompatible/main.go:39:99: cannot use enableAction (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("enable", "Enable previously disabled linters.").PlaceHolder("LINTER").Short('E').Action
../../../../pkg/mod/github.com/alecthomas/gometalinter@v2.0.12+incompatible/main.go:40:83: cannot use cliLinterOverrides (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("linter", "Define a linter.").PlaceHolder("NAME:COMMAND:PATTERN").Action
../../../../pkg/mod/github.com/alecthomas/gometalinter@v2.0.12+incompatible/main.go:43:56: cannot use disableAllAction (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("disable-all", "Disable all linters.").Action
../../../../pkg/mod/github.com/alecthomas/gometalinter@v2.0.12+incompatible/main.go:44:54: cannot use enableAllAction (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.cmdMixin.flagGroup.Flag("enable-all", "Enable all linters.").Action
../../../../pkg/mod/github.com/alecthomas/gometalinter@v2.0.12+incompatible/main.go:200:12: cannot use loadDefaultConfig (type func(*kingpin.Application, *kingpin.ParseElement, *kingpin.ParseContext) error) as type kingpin.Action in argument to app.Action
exit status 2
exit status 1
```

and doesn't intend to: https://github.com/alecthomas/gometalinter/issues/518:

> Closing this because that is no longer a supported installation mechanism. Use `curl https://git.io/vp6lP | sh` or download releases directly.

Also "github.com/opennota/check/cmd/..." fails due to https://github.com/golang/go/issues/27215 - a simple workaround exists.

In addition running `go run build.go setup` updates go.mod/go.sum, which as far as I understand isn't desired.

--- a/build.go
+++ b/build.go
@@ -34,21 +34,22 @@ import (
 )
 
 var (
-	versionRe     = regexp.MustCompile(`-[0-9]{1,3}-g[0-9a-f]{5,10}`)
-	goarch        string
-	goos          string
-	noupgrade     bool
-	version       string
-	goCmd         string
-	goVersion     float64
-	race          bool
-	debug         = os.Getenv("BUILDDEBUG") != ""
-	extraTags     string
-	installSuffix string
-	pkgdir        string
-	cc            string
-	debugBinary   bool
-	timeout       = "120s"
+	versionRe        = regexp.MustCompile(`-[0-9]{1,3}-g[0-9a-f]{5,10}`)
+	goarch           string
+	goos             string
+	noupgrade        bool
+	version          string
+	goCmd            string
+	goVersion        float64
+	race             bool
+	debug            = os.Getenv("BUILDDEBUG") != ""
+	extraTags        string
+	installSuffix    string
+	pkgdir           string
+	cc               string
+	debugBinary      bool
+	timeout          = "120s"
+	gogoProtoVersion = "v1.2.0"
 )
 
 type target struct {
@@ -198,7 +199,7 @@ type dependencyRepo struct {
 }
 
 var dependencyRepos = []dependencyRepo{
-	{path: "protobuf", repo: "https://github.com/gogo/protobuf.git", commit: "v1.2.0"},
+	{path: "protobuf", repo: "https://github.com/gogo/protobuf.git", commit: gogoProtoVersion},
 	{path: "xdr", repo: "https://github.com/calmh/xdr.git", commit: "08e072f9cb16"},
 }
 
@@ -253,9 +254,6 @@ func main() {
 
 func runCommand(cmd string, target target) {
 	switch cmd {
-	case "setup":
-		setup()
-
 	case "install":
 		var tags []string
 		if noupgrade {
@@ -333,33 +331,6 @@ func parseFlags() {
 	flag.StringVar(&cc, "cc", os.Getenv("CC"), "Set CC environment variable for `go build`")
 	flag.BoolVar(&debugBinary, "debug-binary", debugBinary, "Create unoptimized binary to use with delve, set -gcflags='-N -l' and omit -ldflags")
 	flag.Parse()
-}
-
-func setup() {
-	packages := []string{
-		"github.com/alecthomas/gometalinter",
-		"github.com/AlekSi/gocov-xml",
-		"github.com/axw/gocov/gocov",
-		"github.com/FiloSottile/gvt",
-		"golang.org/x/lint/golint",
-		"github.com/gordonklaus/ineffassign",
-		"github.com/mdempsky/unconvert",
-		"github.com/mitchellh/go-wordwrap",
-		"github.com/opennota/check/cmd/...",
-		"github.com/tsenart/deadcode",
-		"golang.org/x/net/html",
-		"golang.org/x/tools/cmd/cover",
-		"honnef.co/go/tools/cmd/gosimple",
-		"honnef.co/go/tools/cmd/staticcheck",
-		"honnef.co/go/tools/cmd/unused",
-		"github.com/josephspurrier/goversioninfo",
-	}
-	for _, pkg := range packages {
-		fmt.Println(pkg)
-		runPrint(goCmd, "get", "-u", pkg)
-	}
-
-	runPrint(goCmd, "install", "-v", "github.com/syncthing/syncthing/vendor/github.com/gogo/protobuf/protoc-gen-gogofast")
 }
 
 func test(pkgs ...string) {
@@ -761,6 +732,7 @@ func shouldRebuildAssets(target, srcdir string) bool {
 }
 
 func proto() {
+	runPrint(goCmd, "get", fmt.Sprintf("github.com/gogo/protobuf/protoc-gen-gogofast@%v", gogoProtoVersion))
 	os.MkdirAll("repos", 0755)
 	for _, dep := range dependencyRepos {
 		path := filepath.Join("repos", dep.path)

